### PR TITLE
Keras support

### PR DESCRIPTION
**Pull request by serach24**

### CODE REVIEW

File: tensorflow/python/keras/engine/base_layer.py [hunk 1]
```diff
@@ -2421,7 +2421,7 @@ def _maybe_cast_inputs(self, inputs, input_list=None):
     compute_dtype_object = self._compute_dtype_object
     should_autocast = (
```
Comment (anandj91) [inline tensorflow/python/keras/engine/base_layer.py:2422]: what is autocast used for? is this part of mixed precision training?

Comment (anandj91) [reply tensorflow/python/keras/engine/base_layer.py:2422]: I think it is safer to cast to cus explicitly as opposed to letting keras to do it automatically.

**Comment by anandj91** [comment]
When you merge this to master, make sure you don't squash all the commits together. Keep the CNN model commit separate from the other two.

File: tensorflow/python/keras/engine/base_layer.py (hunk 1 above)
Comment (serach24) [reply tensorflow/python/keras/engine/base_layer.py:2422]: > is this part of mixed precision training?

yes

Comment (serach24) [reply tensorflow/python/keras/engine/base_layer.py:2422]: > I think it is safer to cast to cus explicitly as opposed to letting keras to do it automatically.

If we manually cast to cus, the logics of loss-scaling is required to be added manually.
